src/error.rs

```diff
@@ -218,141 +227,142 @@ pub(crate) fn map_error(input: rustls::TLSError) -> rustls_result {
 
 impl Display for rustls_result {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", result_to_string(self))
+        let msg: String = match result_to_tlserror(self) {
+            Either::String(s) => s,
+            Either::TLSError(e) => e.to_string(),
+        };
+        write!(f, "{}", msg)
     }
 }
 
-fn result_to_string(input: &rustls_result) -> String {
+// Either a String or a TLSError
+enum Either {
+    String(String),
+    TLSError(TLSError),
+}
+
+// Turn a rustls_result into a TLSError on a best-effort basis. For
+// variants that don't have a corresponding TLSError, or where we want to
+// override TLSError's Display implementation, this returns a String.
+// Otherwise, it returns a TLSError. This is used internally for determining
+// whether a rustls_result is part of some top-level variant that maps to
+// several rustls_results.
+fn result_to_tlserror(input: &rustls_result) -> Either {
     use rustls::internal::msgs::enums::AlertDescription as alert;
     use rustls_result::*;
     use sct::Error as sct;
     use webpki::Error as webpki;
 
-    let msg = match input {
+    match input {
```

Comment:
I like the `Either` enum, but I'm not sure about the two `match` blocks here. It's not great having all the `unreachable!` cases in the second one. Why not one giant `match`? Was it to avoid having dozens of cases that all get wrapped in `Either::TlsError`?

Reply:
Yep, exactly - I didn't want to have to do tons of wrapping.

Reply:
I don't have a suggestion of how to do better so I'll spare you my aesthetic griping. This change definitely achieves what it intends.